### PR TITLE
ARROW-4200: [C++/Python] Enable conda_env_python.yml to work on Windows, simplify python/development.rst

### PR DESCRIPTION
**Pull request by wesm**
I also removed nomkl from conda_env_python.yml. It's sort of a developer decision whether or not they want to install the MKL -- we shouldn't force them to _not_ have it

### CODE REVIEW

File: ci/travis_script_python.sh [hunk 1]
```diff
@@ -47,6 +47,7 @@ fi
 
 conda create -y -q -p $CONDA_ENV_DIR \
       --file $TRAVIS_BUILD_DIR/ci/conda_env_python.yml \
+      nomkl \
```
Comment (kszucs) [inline ci/travis_script_python.sh]: We should add it to the python docker image as well.

Comment (wesm) [reply ci/travis_script_python.sh]: done

**Review by kszucs — APPROVED** [review]

File: docs/source/python/development.rst [hunk 2]
```diff
@@ -310,10 +301,10 @@ First, starting from fresh clones of Apache Arrow:
...
 
-   conda create -y -q -n pyarrow-dev ^
-         python=3.6 numpy six setuptools cython pandas pytest ^
-         cmake flatbuffers rapidjson boost-cpp thrift-cpp snappy zlib ^
-         gflags brotli lz4-c zstd -c conda-forge
+    conda create -y -n pyarrow-dev -c conda-forge ^
+        --file arrow\ci\conda_env_cpp.yml ^
+        --file arrow\ci\conda_env_python.yml ^
+        python=3.7
    activate pyarrow-dev
```
Comment (bkietz) [inline docs/source/python/development.rst]:
```suggestion
   conda activate pyarrow-dev
```

File: docs/source/python/development.rst [hunk 3]
```diff
@@ -86,18 +86,9 @@ On Linux and OSX:
...
-
-.. code-block:: shell
-
-    conda create -y -n pyarrow-dev -c conda-forge ^
-        --file arrow\ci\conda_env_cpp.yml ^
-        --file arrow\ci\conda_env_python.yml ^
-        python=3.6
-
-   activate pyarrow-dev
+For Windows, see the section later on this page.
```
Comment (bkietz) [inline docs/source/python/development.rst]:
```suggestion
For Windows, see the `Developing on Windows`_ section below.
```

**Pull request closed without merging**